src/middlewared/middlewared/etc_files/local/minio/configure.py

```diff
@@ -44,11 +44,16 @@ def render_certificates(s3, middleware):
 
 def configure_minio_sys_dir(s3):
     storage_path = s3['storage_path']
+    # Create storage path if it does not exist
+    os.makedirs(storage_path, exist_ok=True)
```

Comment:
Why should we do that? Do we do that for other share types?

Reply:
@themylogin it's already being done in `s3.update` - similarly it's happening for webdav/smb as well ( Here it's more of a sanity check to ensure it exists complying with `s3.update` ) . We should probably create a ticket to remove this everywhere but that would be change of behavior, so not sure.